test/functionapp.test.ts

```diff
@@ -0,0 +1,125 @@
+        if (!longRunningTestsEnabled) {
+            this.skip();
+        }
+        this.timeout(1200 * 1000);
+        await updateGlobalSetting(projectLanguageSetting, oldProjectLanguage);
+        await vscode.workspace.getConfiguration(extensionPrefix).update('advancedCreation', oldAdvancedCreationSetting, vscode.ConfigurationTarget.Global);
+        const client: ResourceManagementClient = getResourceManagementClient(testAccount);
+        for (const resourceGroup of resourceGroupsToDelete) {
+            if (await client.resourceGroups.checkExistence(resourceGroup)) {
+                console.log(`Deleting resource group "${resourceGroup}"...`);
+                await client.resourceGroups.deleteMethod(resourceGroup);
+                console.log(`Resource group "${resourceGroup}" deleted.`);
+            } else {
+                // If the test failed, the resource group might not actually exist
+                console.log(`Ignoring resource group "${resourceGroup}" because it does not exist.`);
+            }
+        }
+        ext.tree.dispose();
+    });
+
+    test('createFunctionApp (Basic)', async () => {
+        const resourceName2: string = getRandomHexString().toLowerCase();
+        resourceGroupsToDelete.push(resourceName2);
+        await vscode.workspace.getConfiguration(extensionPrefix).update('advancedCreation', false, vscode.ConfigurationTarget.Global);
+        await updateGlobalSetting(projectLanguageSetting, ProjectLanguage.JavaScript);
+        ext.ui = new TestUserInput([resourceName2]);
+        await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(resourceName2, resourceName2);
+        assert.ok(createdApp);
+    });
+
+    test('createFunctionApp (Advanced)', async () => {
+        resourceGroupsToDelete.push(resourceName1);
+        await vscode.workspace.getConfiguration(extensionPrefix).update('advancedCreation', true, vscode.ConfigurationTarget.Global);
+        const testInputs: string[] = [resourceName1, 'Linux', '.NET', '$(plus) Create new resource group', resourceName1, '$(plus) Create new storage account', resourceName1, 'West US'];
+        ext.ui = new TestUserInput(testInputs);
+        await vscode.commands.executeCommand('azureFunctions.createFunctionApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
+        const createdApp: WebSiteManagementModels.Site = await client.webApps.get(resourceName1, resourceName1);
+        assert.ok(createdApp);
+    });
+
+    test('stopFunctionApp', async () => {
+        ext.ui = new TestUserInput([resourceName1]);
+        await vscode.commands.executeCommand('azureFunctions.stopFunctionApp');
+        const client: WebSiteManagementClient = getWebsiteManagementClient(testAccount);
```

Comment:
You need to verify the app is running before you call `azureFunctions.stopFunctionApp`